framework/source/class/qx/ui/table/celleditor/TextField.js

```diff
@@ -38,6 +38,11 @@ qx.Class.define("qx.ui.table.celleditor.TextField",
         value = validationFunc( value, cellEditor.originalValue );
       }
 
+      // Correct problem of NaN displaying when value is null string.
+      if (value == "") {
+          value = null;
+      }
+      
```

Comment:
"" is not null ... so since the NaN problem only occurs if the value is a number, how about moving this inside the 

```javascript
if (typeof cellEditor.originalValue == "number") {
```

block